docs/Xeltalliv/simple3D.md

```diff
@@ -0,0 +1,890 @@
+
+So in short, **this extension does not have any kind of scenes, objects, cameras, etc**, you have to keep track of it yourself.
+
+---
+
+3D models consist of vertices which together form primitives (points, lines, triangles). Each vertex has either 2D (XY) or 3D (XYZ) location described with 2 or 3 numbers respectively. Before drawing the mesh, you would usually set up transformation, which tells how to take those initial locations and transform them to correct location on your 2D screen. The typical way to do it, is to chain multiple simple transformations together. Simple transformations can be translation (offsetting), rotation, scaling, mirroring, skewing, etc.
+
+## Drawing things <a name="simple-drawing"></a>
+For now let's not worry about transformations and just draw something as is.
+First step would be to clear the screen:
+```scratch
+set clear color R: (0.5) G: (0.5) B: (0.5) A: (1) :: sensing
+clear (color and depth v) :: sensing
+```
+After running that the screen should turn gray.
+Now let's draw 2 triangles arranged into a rectangle. First create a mesh:
+```scratch
+create mesh [my mesh] :: sensing
+```
+The create 2 lists `posX` and `posY`.
+Fill `posX` with:
+```
+-0.9
+0.9
+-0.9
+-0.9
+0.9
+0.9
+```
+Fill `posY` with:
+```
+-0.9
+-0.9
+0.9
+0.9
+-0.9
+0.9
+```
+then set mesh's positions to it:
+```scratch
+set [my mesh] positions XY [posX v] [posY v] :: sensing
+```
+and finally draw it:
+```scratch
+draw [my mesh] :: sensing
+```
+The result should look something like this:
+![white rectangle on gray background](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAloAAAHEAgMAAACcsWzXAAABhGlDQ1BJQ0MgcHJvZmlsZQAAKJF9kT1Iw0AcxV/TSkUqImYQcchQnSyCiohTrUIRKoRaoVUH89EvaNKQpLg4Cq4FBz8Wqw4uzro6uAqC4AeIs4OToouU+L+k0CLGg+N+vLv3uHsHcI2KolmhOKDptplOJoRsblUIvyIEHv2YxbikWMacKKbgO77uEWDrXYxl+Z/7c/SqeUsBAgJxXDFMm3iDeHrTNhjvE/NKSVKJz4nHTLog8SPTZY/fGBdd5lgmb2bS88Q8sVDsYLmDlZKpEU8RR1VNp3wu67HKeIuxVqkprXuyF0by+soy02kOI4lFLEGEABk1lFGBjRitOikW0rSf8PEPuX6RXDK5ylDIsYAqNEiuH+wPfndrFSYnvKRIAuh6cZyPESC8CzTrjvN97DjNEyD4DFzpbX+1Acx8kl5va9EjoG8buLhua/IecLkDDD4Zkim5UpAmVygA72f0TTlg4BboWfN6a+3j9AHIUFepG+DgEBgtUva6z7u7O3v790yrvx8KVHLj7WPAsgAAAAlQTFRFNDQ0gICA////76SQuwAAAAlwSFlzAAAYTAAAGHQBn6hAIAAAAAd0SU1FB+gEFBIAAu2UQ3UAAAAZdEVYdENvbW1lbnQAQ3JlYXRlZCB3aXRoIEdJTVBXgQ4XAAABNUlEQVR42u3OUQkAIBAFsCthH0toR01pAX+FB24JVhWqjUTdy8vLy8vLy8vLy8vr99fcEZaXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5eXl5fX7RXGy8vLy8vLy8vLy8vr/SvUAWUTfy76T1mWAAAAAElFTkSuQmCC)
```

Comment:
This image seems to be broken

Reply:
On everything that I tested, seems to work fine for me. It is a png with indexed colors (palette of 3), so that might be the cause.

Reply:
![image broken?](https://github.com/TurboWarp/extensions/assets/137440114/1d335882-0792-4555-be44-4e884d4f5783)

This is how it turns out for me 🤔

Reply:
Oh, that's just github markdown viewer that can't display it:
![image](https://github.com/TurboWarp/extensions/assets/90340988/d7c68086-87ba-4fe7-9698-bc71419afed2)
When it is properly compiled into the TurboWarp docs page by the provided nodejs scripts, it displays fine:
![image](https://github.com/TurboWarp/extensions/assets/90340988/3678d8b0-226c-4023-9785-485cfd427391)

Reply:
I see.